### ISSUE TITLE
[BUG] FutureWarning from Pandas 2.2.1 in groupby call

### ISSUE THREAD

**Issue by bkmartinjr** [scverse/rapids_singlecell#195]
**Describe the bug**
When running the latest version with Pandas 2.2.1 (Python 3.11), a call to `highly_variable_genes` will raise a FutureWarning:

```
/.../miniforge3/envs/rapids_singlecell/lib/python3.11/site-packages/rapids_singlecell/preprocessing/_hvg.py:311: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
disp_grouped = df.groupby("mean_bin")["dispersions"]
```

I believe it just needs an explicitly specified param to resolve this.

**Steps/Code to reproduce bug**

The method was called on an anndata without args, e.g.,
> rsc.pp.highly_variable_genes(adata)

**Expected behavior**

**Environment details (please complete the following information):**
 - Environment location: bare metal
 - Linux Distro/Architecture: Ubuntu 22.04 amd64
 - GPU Model/Driver: GA107M (RTX 3050)
 - CUDA: 11.8
 - Method of Rapids install:  mamba

```
pandas                    2.2.1           py311h320fe9a_0    conda-forge
rapids                    24.04.00        cuda11_py311_240411_gb301c96_0    rapidsai
rapids-singlecell         0.10.2                   pypi_0    pypi
```

**Comment by Intron7** [comment]
@bkmartinjr Thank you so much for reporting this. I'll test some fix and make a PR soon.

**Comment by Intron7** [comment]
I just checked. This got fixed with version 0.10.3

**Comment by bkmartinjr** [comment]
Confirmed!  Thank you for a fantastic package.
